doc/source/glossary.rst

```diff
@@ -0,0 +1,179 @@
+Glossary
+========
+There are many terms when talking about memory forensics, this list hopes to define the common ones and
+provide some commonality on how to refer to particular ideas within the field.
+
+A
+-
+.. _Address:
+    An address is another name for an :ref:`offset<Offset>`, specifically an offset within memory.  Offsets can be
+    both relative or absolute, whereas addresses are almost always absolute.
+
+.. _Address Space:
+
+Address Space
+    This is the name in volatility 2 for what's referred to as a :ref:`Translation Layer<translation layer>`.  It
+    encompasses all values that can be addresses, usually in reference to addresses in memory.
+
+.. _Alignment:
+
+Alignment
+    This value is what all data :ref:`offsets<offset>` will typically be a multiple of within a :ref:`type<type>`.
+
+.. _Array:
+
+Array
+    This represents a list of items, which can be access by an index, which is zero-based (meaning the first
+    element has index 0).  Items in arrays are almost always the same size (it is not a generic list, as in python)
+    even if they are :ref:`pointers<pointer>` to different sized objects.
+
+D
+-
+.. _Data Layer:
+
+Data Layer
+    A group of bytes, where each byte can be addressed by a specific offset.  Data layers are usually contiguous
+    chunks of data.
+
+.. _Dereference:
+
+Dereference
+    The act of taking the value of a pointer, and using it as an offset to another object, as a reference.
+
+.. _Domain:
+
+Domain
+    This the grouping for input values for a mapping or mathematical function.
+
+M
+-
+.. _Map:
+
+Map, mapping
+    A mapping is a relationship between two values, where one value (the :ref:`Domain<domain>` maps to the :ref:`Range<range>` value).
```

Comment:
Suggestion: 
> ...is a relationship between two sets. Where elements of the Domain map to the elements of the Range.
Compare: https://en.wikipedia.org/wiki/Function_(mathematics) (Found under: https://en.wikipedia.org/wiki/Mapping)

Reply:
Yeah, I tend to see it more in terms of an input and an output, but mathematically you're right it's a relationship between two sets.  Updated to reflect this.  5:)